### PR TITLE
Music: update README

### PR DESCRIPTION
**Pull request by sanchitmalhotra126**
Some README updates capturing our execution and player logic/architecture, and high level overview of view components.

### CODE REVIEW

File: apps/src/music/README.md [hunk 1]
```diff
@@ -6,7 +6,8 @@
 
 This is the client-side code for **Music Lab**, as instantiated by a `music` level using [Lab2](../lab2/).
 
-**Music Lab** is a powerful lab for creating music using code.  It includes the following features:
+**Music Lab** is a powerful lab for creating music using code. It includes the following features:
```
Comment (sanchitmalhotra126) [inline apps/src/music/README.md:9]: The first few changes are just editor auto-formatting; the new content begins with `## Compilation, Execution, and Playback`

File: apps/src/music/README.md [hunk 2]
```diff
@@ -31,30 +32,71 @@ This is the client-side code for **Music Lab**, as instantiated by a `music` lev
...
+
+### Handling Updates Mid-Playback
+
+Music Lab supports live updates during playback. When code changes are detected, code is recompiled and executed in the same manner as above. The new list of PlaybackEvents is handed to the player, along with a flag that indicates to the player that it should replace its current set of events with this new set. The player cancels all pending events, and requeues the new set of events. Depending on the specific player implementation, the user hears the updates almost immediately, or latest when the next sound starts.
+
+### Triggered Events
+
+During compliation, code attached to trigger blocks is also compiled. When the trigger is pressed, the trigger's pre-compiled code is executed with a reset Sequencer, which assembles the list of PlaybackEvents for the trigger code specifically. These events are then passed to the player which schedules them alongside the already ongoing events. If code in a trigger block is changed mid-playback, the events will be recompiled and executed as above, and rescheduled from the time the trigger had been pressed.
```
Comment (cnbrenci) [inline apps/src/music/README.md]: typo

**Review by cnbrenci — APPROVED** [review]
Great to see docs near code! Thanks for adding this.

File: apps/src/music/README.md (hunk 2 above)
Comment (cnbrenci) [reply apps/src/music/README.md]: oh it didn't keep my highlight - "compliation" -> "compilation"

File: apps/src/music/README.md (hunk 1 above)
Comment (breville) [reply apps/src/music/README.md:9]: I still write with two spaces after a period for almost everything. :)

File: apps/src/music/README.md [hunk 3]
```diff
@@ -31,30 +32,71 @@ This is the client-side code for **Music Lab**, as instantiated by a `music` lev
...
+
+### Handling Updates Mid-Playback
+
+Music Lab supports live updates during playback. When code changes are detected, code is recompiled and executed in the same manner as above. The new list of PlaybackEvents is handed to the player, along with a flag that indicates to the player that it should replace its current set of events with this new set. The player cancels all pending events, and requeues the new set of events. Depending on the specific player implementation, the user hears the updates almost immediately, or latest when the next sound starts.
```
Comment (breville) [inline apps/src/music/README.md]: > or latest when the next sound starts

later?

File: apps/src/music/README.md [hunk 4]
```diff
@@ -31,30 +32,71 @@ This is the client-side code for **Music Lab**, as instantiated by a `music` lev
...
+
+- [PlaybackEvents](./player/interfaces/PlaybackEvent.ts) represent single items in the timeline. These are used by both UI components and high-level player components like the [Sequencer](./player/sequencer/Sequencer.ts) and [MusicPlayer](./player/MusicPlayer.ts). The playback time for a PlaybackEvent is 1-based, corresponding the number of measures where it starts (ex. 1.5 refers to half a measure after measure 1, or measure 1 beat 3). There are a few different types:
+
+  - [SoundEvents](./player/interfaces/SoundEvent.ts) represent a sound generated by the "play sound" block.
+  - [ChordEvents](./player/interfaces/ChordEvent.ts) represent a chord/set of notes generated by the "play notes" block.
+  - [PatternEvents](./player/interfaces/PatternEvent.ts) represent a drum pattern generated by the "play drums" block.
+
+- [SampleEvents](./player/types.ts#L100) represent a single event to be played by either of the internal player implementations, and therefore used in the [AudioPlayer](./player/types.ts#L8) interface. The MusicPlayer converts PlaybackEvents to SampleEvents. The playback time for a SampleEvent is also 1-based, corresponding the number of measures.
```
Comment (breville) [inline apps/src/music/README.md]: > corresponding the number of measures

corresponding to?

File: apps/src/music/README.md [hunk 5]
```diff
@@ -31,30 +32,71 @@ This is the client-side code for **Music Lab**, as instantiated by a `music` lev
...
+  - The [ToneJSPlayer](./player/ToneJSPlayer.ts) internally converts the the playback time to a "transport time" used by ToneJS APIs. This is similar, but instead expressed as a string in the format "bars:beats:sixteenths" and 0-based. For example, a playback time of 1.5 would be transport time of "0:2:0".
+  - The [SamplePlayer](./player/SamplePlayer.ts) internally converts the playback time to a millisecond offset time from the play start time. This offset time is based on the project's BPM.
+- [SamplerSequences](./player/types.ts#L122) similarly represent a set of notes to be played by a sampler instrument. Currently, only the ToneJSPlayer supports playing SamplerSequences (this is indicated by the [supportsSamplers()](./player/types.ts#L10) function on the AudioPlayer interface).
+
+## View Components
+
+- [MusicView](./views/MusicView.jsx) is the entrypoint into Music Lab, and is top-level container view that is rendered by the Lab2 framework. MusicView handles orchestrating a lot of the overarching app logic, such as connecting UI events to player APIs and loading the music library.
```
Comment (breville) [inline apps/src/music/README.md]: > and is top-level container view

the top-level?

**Review by breville — APPROVED** [review]
This is great!

File: apps/src/music/README.md (hunk 3 above)
Comment (sanchitmalhotra126) [reply apps/src/music/README.md]: Yeah this is worded weirdly, I'll just remove - meaning to say that with the ToneJSPlayer, updates are heard immediately and with the SamplePlayer, updates are heard when the next sounds starts.

**Pull request merged**